quisp/modules/ConnectionManager.cc

```diff
@@ -198,153 +195,159 @@ void ConnectionManager::storeRuleSetForApplication(ConnectionSetupResponse *pk)
  * \todo Always room to make this better.  Ideally should be
  * a _configurable choice_, or even a _policy_ implementation.
  */
-void ConnectionManager::responder_alloc_req_handler(ConnectionSetupRequest *pk) {
+void ConnectionManager::respondToRequest(ConnectionSetupRequest *req) {
   // Taking qnic information of responder node.
-  int actual_dst = pk->getActual_destAddr();
-  int actual_src = pk->getActual_srcAddr();  // initiator address (to get input qnic)
-  int local_qnic_address_to_actual_dst = routing_daemon->return_QNIC_address_to_destAddr(actual_dst);  // This must be -1
-  int local_qnic_address_to_actual_src = routing_daemon->return_QNIC_address_to_destAddr(actual_src);  // TODO: premise only oneconnection allowed btw, two nodes.
+  int actual_dst = req->getActual_destAddr();
+  int actual_src = req->getActual_srcAddr();  // initiator address (to get input qnic)
+
+  // This must be -1
+  int local_qnic_address_to_actual_dst = routing_daemon->return_QNIC_address_to_destAddr(actual_dst);
+  if (local_qnic_address_to_actual_dst != -1) {
+    error("something error happen!");
+  }
+
+  // TODO: premise only one connection allowed btw, two nodes.
+  int local_qnic_address_to_actual_src = routing_daemon->return_QNIC_address_to_destAddr(actual_src);
+  if (local_qnic_address_to_actual_src == -1) {
+    error("This shouldn't happen!");
+  }
+
   connection_setup_inf dst_inf = hardware_monitor->return_setupInf(local_qnic_address_to_actual_dst);
```

Comment:
In this class, we have many similar procedures that retrieving `connection_setup_inf` by actual node addr.
we can combine these procedures into one method and reuse it.
but `HardwareMonitor::return_setupInf` returns actual value not ptr for that, and modifying it needs a big surgery.

I'll fix `HardwareMonitor::return_setupInf` to return `connection_setup_inf*` and then refactor ConnectionManager again in other PR.